File: easy/1790_check_if_one_string_swap_can_make_strings_equal/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  bool areAlmostEqual(std::string s1, std::string s2) {
    std::vector<int> difference;

    for (int i = 0; i < s1.size(); ++i) {
      if (s1[i] != s2[i])
        difference.push_back(i);
    }

    return difference.size() == 0 ||
           (difference.size() == 2 && s1[difference[0]] == s2[difference[1]] &&
            s1[difference[1]] == s2[difference[0]]);
  }
};

File: easy/1790_check_if_one_string_swap_can_make_strings_equal/solution.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        difference = [i for i in range(len(s1)) if s1[i] != s2[i]]

        return len(difference) == 0 or (
            len(difference) == 2
            and s1[difference[0]] == s2[difference[1]]
            and s1[difference[1]] == s2[difference[0]]
        )
